store/SPEC_STANDARD.md

```diff
@@ -0,0 +1,110 @@
+## What is a store specification document?
```

Comment:
Do you think it should be generalized? We do have this too, and it could potentially be extended https://docs.cosmos.network/main/spec/SPEC-SPEC

Reply:
module spec and core level specs should be different. The linked spec template is more module level specs. Maybe we should update that to mention that? 

I think the spec here should be a core level spec template instead of only store.

Reply:
That makes senses. Then they could be both under docs/spec too, so it renders in the docs.

Reply:
That sounds good. I'll do a pass and make sure I remove store-specific text. Apart from the store package, what's considered core (anything out of the modules folder?)

Also, any reason why we want to have a separate one for modules? what's specific there?

Reply:
modules have messages, events and specific things for interaction with the module. I think what we are calling a spec in modules is actually documentation on the module more than the spec. We could reuse this for module rewrites in the future.

Reply:
> We could reuse this for module rewrites in the future.

Sounds good.